##### Importing Required LIbraries

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
# Loading the data into data frame
df = pd.read_excel('Cleaned Data.xlsx')

In [3]:
df.head()

I am currently employed at least part-time  \
0                                           0   
1                                           1   
2                                           1   
3                                           0   
4                                           1   

   I identify as having a mental illness                Education  \
0                                      0       High School or GED   
1                                      1                 Some Phd   
2                                      0  Completed Undergraduate   
3                                      0       Some Undergraduate   
4                                      1  Completed Undergraduate   

   I have my own computer separate from a smart phone  \
0                                                  0    
1                                                  1    
2                                                  1    
3                                                  1    
4                                                  1    

   I have been hospitalized before for my mental illness  \
0                                                  0       
1                                                  0       
2                                                  0       
3                                                  0       
4                                                  1       

   How many days were you hospitalized for your mental illness  \
0                                                0.0             
1                                                0.0             
2                                                0.0             
3                                                NaN             
4                                               35.0             

   I am legally disabled  I have my regular access to the internet  \
0                      0                                         1   
1                      0                                         1   
2                      0                                         1   
3                      0                                         1   
4                      1                                         1   

   I live with my parents  I have a gap in my resume  ...  Obsessive thinking  \
0                       0                          1  ...                 1.0   
1                       0                          0  ...                 0.0   
2                       0                          0  ...                 0.0   
3                       1                          1  ...                 0.0   
4                       0                          1  ...                 1.0   

   Mood swings  Panic attacks  Compulsive behavior  Tiredness    Age  Gender  \
0          0.0            1.0                  0.0        0.0  30-44    Male   
1          0.0            1.0                  0.0        1.0  18-29    Male   
2          0.0            0.0                  0.0        0.0  30-44    Male   
3          0.0            0.0                  0.0        0.0  30-44    Male   
4          1.0            1.0                  1.0        1.0  30-44    Male   

    Household Income              Region               Device Type  
0    $25,000-$49,999            Mountain    Android Phone / Tablet  
1    $50,000-$74,999  East South Central    MacOS Desktop / Laptop  
2  $150,000-$174,999             Pacific    MacOS Desktop / Laptop  
3    $25,000-$49,999         New England  Windows Desktop / Laptop  
4    $25,000-$49,999  East North Central        iOS Phone / Tablet  

[5 rows x 31 columns]

##### Identifying Missing Values

In [4]:
missing_data = df.isnull()
missing_data.head(5)

I am currently employed at least part-time  \
0                                       False   
1                                       False   
2                                       False   
3                                       False   
4                                       False   

   I identify as having a mental illness  Education  \
0                                  False      False   
1                                  False      False   
2                                  False      False   
3                                  False      False   
4                                  False      False   

   I have my own computer separate from a smart phone  \
0                                              False    
1                                              False    
2                                              False    
3                                              False    
4                                              False    

   I have been hospitalized before for my mental illness  \
0                                              False       
1                                              False       
2                                              False       
3                                              False       
4                                              False       

   How many days were you hospitalized for your mental illness  \
0                                              False             
1                                              False             
2                                              False             
3                                               True             
4                                              False             

   I am legally disabled  I have my regular access to the internet  \
0                  False                                     False   
1                  False                                     False   
2                  False                                     False   
3                  False                                     False   
4                  False                                     False   

   I live with my parents  I have a gap in my resume  ...  Obsessive thinking  \
0                   False                      False  ...               False   
1                   False                      False  ...               False   
2                   False                      False  ...               False   
3                   False                      False  ...               False   
4                   False                      False  ...               False   

   Mood swings  Panic attacks  Compulsive behavior  Tiredness    Age  Gender  \
0        False          False                False      False  False   False   
1        False          False                False      False  False   False   
2        False          False                False      False  False   False   
3        False          False                False      False  False   False   
4        False          False                False      False  False   False   

   Household Income  Region  Device Type  
0             False   False        False  
1             False   False        False  
2             False   False        False  
3             False   False        False  
4             False   False        False  

[5 rows x 31 columns]

In [5]:
# Counting missing values in each column
for column in missing_data.columns.values.tolist():
    print(column)
    print (missing_data[column].value_counts())
    print("")    

I am currently employed at least part-time
False    334
Name: I am currently employed at least part-time, dtype: int64

I identify as having a mental illness
False    334
Name: I identify as having a mental illness, dtype: int64

Education
False    334
Name: Education, dtype: int64

I have my own computer separate from a smart phone
False    334
Name: I have my own computer separate from a smart phone, dtype: int64

I have been hospitalized before for my mental illness
False    334
Name: I have been hospitalized before for my mental illness, dtype: int64

How many days were you hospitalized for your mental illness
False    297
True      37
Name: How many days were you hospitalized for your mental illness, dtype: int64

I am legally disabled
False    334
Name: I am legally disabled, dtype: int64

I have my regular access to the internet
False    334
Name: I have my regular access to the internet, dtype: int64

I live with my parents
False    334
Name: I live with my parents, dtype: int6

Columns that contains missing values.
1. How many days were you hospitalized for your mental illness: 37 NaN values
2. Lack of concentration: 1 NaN value
3. Obsessive thinking: 1 NaN value
4. Mood swings: 1 NaN value
5. Panic attacks: 1 NaN value
6. Compulsive behavior: 1 NaN value
7. Tiredness: 1 NaN value
8. Region: 2 NaN value

Dropping the NaN values will not a good approach because the data set contains 334 data points.

In [6]:
# How many days were you hospitalized for your mental illness: 37 NaN values replacing with mean.
avg_days = df["How many days were you hospitalized for your mental illness"].astype("float").mean(axis=0)
print("Average of normalized-losses:", avg_days)

Average of normalized-losses: 3.276094276094276


In [7]:
avg_days = round(3.276094,0)

In [8]:
avg_days

3.0

In [9]:
df["How many days were you hospitalized for your mental illness"].replace(np.nan, avg_days, inplace=True)

In [10]:
df["How many days were you hospitalized for your mental illness"].isnull().sum()

0

Great no NaN values left.

In [11]:
# 2 Lack of concentration: 1 NaN value
df["Lack of concentration"].value_counts()

0.0    282
1.0     51
Name: Lack of concentration, dtype: int64

In [12]:
# already contains maximum numbers of 0, replacing the nan value with 1
df["Lack of concentration"].replace(np.nan, 1, inplace=True)

In [13]:
df["Lack of concentration"].value_counts()

0.0    282
1.0     52
Name: Lack of concentration, dtype: int64

In [14]:
df["Lack of concentration"].isnull().sum()

0

Great no NaN values

In [15]:
# 3 Obsessive thinking: 1 NaN value
df["Obsessive thinking"].value_counts()

0.0    291
1.0     42
Name: Obsessive thinking, dtype: int64

In [16]:
# already contains maximum numbers of 0, replacing the nan value with 1
df["Obsessive thinking"].replace(np.nan, 1, inplace=True)

In [17]:
df["Obsessive thinking"].value_counts()

0.0    291
1.0     43
Name: Obsessive thinking, dtype: int64

In [18]:
df["Obsessive thinking"].isnull().sum()

0

Great no NaN values

In [19]:
# 4 Mood swings: 1 NaN value
df["Mood swings"].value_counts()

0.0    295
1.0     38
Name: Mood swings, dtype: int64

In [20]:
# already contains maximum numbers of 0, replacing the nan value with 1
df["Mood swings"].replace(np.nan, 1, inplace=True)

In [21]:
df["Mood swings"].value_counts()

0.0    295
1.0     39
Name: Mood swings, dtype: int64

In [22]:
df["Obsessive thinking"].isnull().sum()

0

Great no NaN values

In [23]:
# 5 Panic attacks: 1 NaN value
df["Panic attacks"].value_counts()

0.0    284
1.0     49
Name: Panic attacks, dtype: int64

In [24]:
# already contains maximum numbers of 0, replacing the nan value with 1
df["Panic attacks"].replace(np.nan, 1, inplace=True)

In [25]:
df["Panic attacks"].value_counts()

0.0    284
1.0     50
Name: Panic attacks, dtype: int64

In [26]:
df["Panic attacks"].isnull().sum()

0

Great no NaN Values

In [27]:
# 6 Compulsive behavior: 1 NaN value
df["Compulsive behavior"].value_counts()

0.0    304
1.0     29
Name: Compulsive behavior, dtype: int64

In [28]:
# already contains maximum numbers of 0, replacing the nan value with 1
df["Compulsive behavior"].replace(np.nan, 1, inplace=True)

In [29]:
df["Compulsive behavior"].value_counts()

0.0    304
1.0     30
Name: Compulsive behavior, dtype: int64

In [30]:
df["Compulsive behavior"].isnull().sum()

0

Great no NaN values

In [31]:
# 7 Tiredness: 1 NaN value
df["Tiredness"].value_counts()

0.0    233
1.0    100
Name: Tiredness, dtype: int64

In [32]:
# already contains maximum numbers of 0, replacing the nan value with 1
df["Tiredness"].replace(np.nan, 1, inplace=True)

In [33]:
df["Tiredness"].isnull().sum()

0

Great no NaN values

In [34]:
# 8 Region: 2 NaN value
df["Region"].value_counts()

South Atlantic        63
Middle Atlantic       56
East North Central    50
Pacific               46
Mountain              32
West South Central    32
New England           21
East South Central    19
West North Central    13
Name: Region, dtype: int64

In [35]:
# already contains minimum of West North Central, replacing the nan value with West North Central
df["Region"].replace(np.nan,'West North Central' , inplace=True)

In [36]:
df["Region"].isnull().sum()

0

Great no NaN values

In [37]:
df.isnull().sum()

I am currently employed at least part-time                      0
I identify as having a mental illness                           0
Education                                                       0
I have my own computer separate from a smart phone              0
I have been hospitalized before for my mental illness           0
How many days were you hospitalized for your mental illness     0
I am legally disabled                                           0
I have my regular access to the internet                        0
I live with my parents                                          0
I have a gap in my resume                                       0
Total length of any gaps in my resume in months.                0
Annual income (including any social welfare programs) in USD    0
I am unemployed                                                 0
I read outside of work and school                               0
Annual income from social welfare programs                      0
I receive 

Well whole data is free from NaN values.

###### Renaming the coulmns for better readability, Correcting Data format, handling categorical values.

1. Renaming the columns

In [38]:
df.columns

Index(['I am currently employed at least part-time',
       'I identify as having a mental illness', 'Education',
       'I have my own computer separate from a smart phone',
       'I have been hospitalized before for my mental illness',
       'How many days were you hospitalized for your mental illness',
       'I am legally disabled', 'I have my regular access to the internet',
       'I live with my parents', 'I have a gap in my resume',
       'Total length of any gaps in my resume in months.',
       'Annual income (including any social welfare programs) in USD',
       'I am unemployed', 'I read outside of work and school',
       'Annual income from social welfare programs', 'I receive food stamps',
       'I am on section 8 housing',
       'How many times were you hospitalized for your mental illness',
       'Lack of concentration', 'Anxiety', 'Depression', 'Obsessive thinking',
       'Mood swings', 'Panic attacks', 'Compulsive behavior', 'Tiredness',
       'Age', 'Gender

In [39]:
# Renaming the columns
df = df.rename(columns={'I am currently employed at least part-time':'part-time-employed','I identify as having a mental illness':'mental-illness','I have my own computer separate from a smart phone':'computer',
                       'I have been hospitalized before for my mental illness':'hospitalized-MI','How many days were you hospitalized for your mental illness':'hospitalized-days',
                       'I am legally disabled':'legally-disabled','I have my regular access to the internet':'internet-access','I live with my parents':'live-with-parents',
                       'I have a gap in my resume':'gap-in-resume','Total length of any gaps in my resume in months.':'resume-gap-in-months',
                       'Annual income (including any social welfare programs) in USD':'total-annual-income (USD)','I am unemployed':'unemployed',
                       'I read outside of work and school':'read-outside','Annual income from social welfare programs':'annual-income-SocialWelfare',
                       'I receive food stamps':'recieve-food-stamps','I am on section 8 housing':'housing-section-8',
                       'How many times were you hospitalized for your mental illness':'hospitalized-for-MI(num.of times)'})

In [40]:
df.columns

Index(['part-time-employed', 'mental-illness', 'Education', 'computer',
       'hospitalized-MI', 'hospitalized-days', 'legally-disabled',
       'internet-access', 'live-with-parents', 'gap-in-resume',
       'Total length of any gaps in my resume in months.',
       'total-annual-income (USD)', 'unemployed', 'read-outside',
       'annual-income-SocialWelfare', 'recieve-food-stamps',
       'housing-section-8', 'hospitalized-for-MI(num.of times)',
       'Lack of concentration', 'Anxiety', 'Depression', 'Obsessive thinking',
       'Mood swings', 'Panic attacks', 'Compulsive behavior', 'Tiredness',
       'Age', 'Gender', 'Household Income', 'Region', 'Device Type'],
      dtype='object')

In [41]:
df.head()

part-time-employed  mental-illness                Education  computer  \
0                   0               0       High School or GED         0   
1                   1               1                 Some Phd         1   
2                   1               0  Completed Undergraduate         1   
3                   0               0       Some Undergraduate         1   
4                   1               1  Completed Undergraduate         1   

   hospitalized-MI  hospitalized-days  legally-disabled  internet-access  \
0                0                0.0                 0                1   
1                0                0.0                 0                1   
2                0                0.0                 0                1   
3                0                3.0                 0                1   
4                1               35.0                 1                1   

   live-with-parents  gap-in-resume  ...  Obsessive thinking  Mood swings  \
0                  0              1  ...                 1.0          0.0   
1                  0              0  ...                 0.0          0.0   
2                  0              0  ...                 0.0          0.0   
3                  1              1  ...                 0.0          0.0   
4                  0              1  ...                 1.0          1.0   

   Panic attacks  Compulsive behavior  Tiredness    Age  Gender  \
0            1.0                  0.0        0.0  30-44    Male   
1            1.0                  0.0        1.0  18-29    Male   
2            0.0                  0.0        0.0  30-44    Male   
3            0.0                  0.0        0.0  30-44    Male   
4            1.0                  1.0        1.0  30-44    Male   

    Household Income              Region               Device Type  
0    $25,000-$49,999            Mountain    Android Phone / Tablet  
1    $50,000-$74,999  East South Central    MacOS Desktop / Laptop  
2  $150,000-$174,999             Pacific    MacOS Desktop / Laptop  
3    $25,000-$49,999         New England  Windows Desktop / Laptop  
4    $25,000-$49,999  East North Central        iOS Phone / Tablet  

[5 rows x 31 columns]

##### Handling Categorical Data

###### 1. Education

In [42]:
df['Education'].value_counts()

Completed Undergraduate    100
Some Undergraduate          82
High School or GED          63
Completed Masters           49
Some Masters                12
Completed Phd               10
Some highschool             10
Some Phd                     8
Name: Education, dtype: int64

In [43]:
dummy_education = pd.get_dummies(df['Education'])
dummy_education.head()

Completed Masters  Completed Phd  Completed Undergraduate  \
0                  0              0                        0   
1                  0              0                        0   
2                  0              0                        1   
3                  0              0                        0   
4                  0              0                        1   

   High School or GED  Some Phd  Some Undergraduate  Some highschool  \
0                   1         0                   0                0   
1                   0         1                   0                0   
2                   0         0                   0                0   
3                   0         0                   1                0   
4                   0         0                   0                0   

   Some Masters  
0             0  
1             0  
2             0  
3             0  
4             0

In [44]:
dummy_education.rename(columns={'Completed Masters':'edu-completed-masters', 'Completed Phd':'edu-completed-phd',
                               'Completed Undergraduate':'edu-completed-undergraduate','High School or GED':'edu-high-school/ged',
                               'Some Phd':'edu-some-phd','Some Undergraduate':'edu-some-undergraduate','Some highschool':'edu-some-highschool',
                               'Some Masters':'edu-some-masters'}, inplace=True)
dummy_education.head()

edu-completed-masters  edu-completed-phd  edu-completed-undergraduate  \
0                      0                  0                            0   
1                      0                  0                            0   
2                      0                  0                            1   
3                      0                  0                            0   
4                      0                  0                            1   

   edu-high-school/ged  edu-some-phd  edu-some-undergraduate  \
0                    1             0                       0   
1                    0             1                       0   
2                    0             0                       0   
3                    0             0                       1   
4                    0             0                       0   

   edu-some-highschool  Some Masters  
0                    0             0  
1                    0             0  
2                    0             0  
3                    0             0  
4                    0             0

In [45]:
# concatinating 
df = pd.concat([df,dummy_education], axis=1)
# dropping original column aspiration
df.drop('Education', axis =1, inplace= True)

In [46]:
df.head()

part-time-employed  mental-illness  computer  hospitalized-MI  \
0                   0               0         0                0   
1                   1               1         1                0   
2                   1               0         1                0   
3                   0               0         1                0   
4                   1               1         1                1   

   hospitalized-days  legally-disabled  internet-access  live-with-parents  \
0                0.0                 0                1                  0   
1                0.0                 0                1                  0   
2                0.0                 0                1                  0   
3                3.0                 0                1                  1   
4               35.0                 1                1                  0   

   gap-in-resume  Total length of any gaps in my resume in months.  ...  \
0              1                                                24  ...   
1              0                                                 1  ...   
2              0                                                 0  ...   
3              1                                                11  ...   
4              1                                                33  ...   

               Region               Device Type  edu-completed-masters  \
0            Mountain    Android Phone / Tablet                      0   
1  East South Central    MacOS Desktop / Laptop                      0   
2             Pacific    MacOS Desktop / Laptop                      0   
3         New England  Windows Desktop / Laptop                      0   
4  East North Central        iOS Phone / Tablet                      0   

   edu-completed-phd  edu-completed-undergraduate  edu-high-school/ged  \
0                  0                            0                    1   
1                  0                            0                    0   
2                  0                            1                    0   
3                  0                            0                    0   
4                  0                            1                    0   

   edu-some-phd  edu-some-undergraduate  edu-some-highschool  Some Masters  
0             0                       0                    0             0  
1             1                       0                    0             0  
2             0                       0                    0             0  
3             0                       1                    0             0  
4             0                       0                    0             0  

[5 rows x 38 columns]

###### 2. Region

In [47]:
df['Region'].value_counts()

South Atlantic        63
Middle Atlantic       56
East North Central    50
Pacific               46
Mountain              32
West South Central    32
New England           21
East South Central    19
West North Central    15
Name: Region, dtype: int64

In [48]:
dummy_region = pd.get_dummies(df['Region'])
dummy_region.head()

East North Central  East South Central  Middle Atlantic  Mountain  \
0                   0                   0                0         1   
1                   0                   1                0         0   
2                   0                   0                0         0   
3                   0                   0                0         0   
4                   1                   0                0         0   

   New England  Pacific  South Atlantic  West North Central  \
0            0        0               0                   0   
1            0        0               0                   0   
2            0        1               0                   0   
3            1        0               0                   0   
4            0        0               0                   0   

   West South Central  
0                   0  
1                   0  
2                   0  
3                   0  
4                   0

In [49]:
dummy_region.rename(columns={'East North Central':'region-east-north-central', 'East South Central':'region-east-south-central',
                               'Middle Atlantic':'region-middle-atlantic','Mountain':'region-mountain',
                               'New England':'region-new-england','Pacific':'region-pacific','South Atlantic':'region-south-atlantic',
                               'West North Central':'region-west-north-central','West South Central':'region-west-south-central'}, inplace=True)
dummy_region.head()

region-east-north-central  region-east-south-central  \
0                          0                          0   
1                          0                          1   
2                          0                          0   
3                          0                          0   
4                          1                          0   

   region-middle-atlantic  region-mountain  region-new-england  \
0                       0                1                   0   
1                       0                0                   0   
2                       0                0                   0   
3                       0                0                   1   
4                       0                0                   0   

   region-pacific  region-south-atlantic  region-west-north-central  \
0               0                      0                          0   
1               0                      0                          0   
2               1                      0                          0   
3               0                      0                          0   
4               0                      0                          0   

   region-west-south-central  
0                          0  
1                          0  
2                          0  
3                          0  
4                          0

In [50]:
# concatinating 
df = pd.concat([df,dummy_region], axis=1)
# dropping original column aspiration
df.drop('Region', axis =1, inplace= True)

In [51]:
df.head()

part-time-employed  mental-illness  computer  hospitalized-MI  \
0                   0               0         0                0   
1                   1               1         1                0   
2                   1               0         1                0   
3                   0               0         1                0   
4                   1               1         1                1   

   hospitalized-days  legally-disabled  internet-access  live-with-parents  \
0                0.0                 0                1                  0   
1                0.0                 0                1                  0   
2                0.0                 0                1                  0   
3                3.0                 0                1                  1   
4               35.0                 1                1                  0   

   gap-in-resume  Total length of any gaps in my resume in months.  ...  \
0              1                                                24  ...   
1              0                                                 1  ...   
2              0                                                 0  ...   
3              1                                                11  ...   
4              1                                                33  ...   

   Some Masters  region-east-north-central  region-east-south-central  \
0             0                          0                          0   
1             0                          0                          1   
2             0                          0                          0   
3             0                          0                          0   
4             0                          1                          0   

   region-middle-atlantic  region-mountain  region-new-england  \
0                       0                1                   0   
1                       0                0                   0   
2                       0                0                   0   
3                       0                0                   1   
4                       0                0                   0   

   region-pacific  region-south-atlantic  region-west-north-central  \
0               0                      0                          0   
1               0                      0                          0   
2               1                      0                          0   
3               0                      0                          0   
4               0                      0                          0   

   region-west-south-central  
0                          0  
1                          0  
2                          0  
3                          0  
4                          0  

[5 rows x 46 columns]

###### 3. Device Type

In [52]:
df['Device Type'].value_counts()

Windows Desktop / Laptop    122
Android Phone / Tablet       93
iOS Phone / Tablet           93
MacOS Desktop / Laptop       24
Other                         2
Name: Device Type, dtype: int64

In [53]:
dummy_DT = pd.get_dummies(df['Device Type'])
dummy_DT.head()

Android Phone / Tablet  MacOS Desktop / Laptop  Other  \
0                       1                       0      0   
1                       0                       1      0   
2                       0                       1      0   
3                       0                       0      0   
4                       0                       0      0   

   Windows Desktop / Laptop  iOS Phone / Tablet  
0                         0                   0  
1                         0                   0  
2                         0                   0  
3                         1                   0  
4                         0                   1

In [54]:
dummy_DT.rename(columns={'Android Phone / Tablet':'DT-Android Phone / Tablet', 'MacOS Desktop / Laptop':'DT-MacOS Desktop / Laptop',
                               'Other':'DT-Other','Windows Desktop / Laptop':'DT-Windows Desktop / Laptop',
                               'iOS Phone / Tablet':'DT-iOS Phone / Tablet'}, inplace=True)
dummy_DT.head()

DT-Android Phone / Tablet  DT-MacOS Desktop / Laptop  DT-Other  \
0                          1                          0         0   
1                          0                          1         0   
2                          0                          1         0   
3                          0                          0         0   
4                          0                          0         0   

   DT-Windows Desktop / Laptop  DT-iOS Phone / Tablet  
0                            0                      0  
1                            0                      0  
2                            0                      0  
3                            1                      0  
4                            0                      1

In [55]:
# concatinating 
df = pd.concat([df,dummy_DT], axis=1)
# dropping original column aspiration
df.drop('Device Type', axis =1, inplace= True)

In [56]:
df.head()

part-time-employed  mental-illness  computer  hospitalized-MI  \
0                   0               0         0                0   
1                   1               1         1                0   
2                   1               0         1                0   
3                   0               0         1                0   
4                   1               1         1                1   

   hospitalized-days  legally-disabled  internet-access  live-with-parents  \
0                0.0                 0                1                  0   
1                0.0                 0                1                  0   
2                0.0                 0                1                  0   
3                3.0                 0                1                  1   
4               35.0                 1                1                  0   

   gap-in-resume  Total length of any gaps in my resume in months.  ...  \
0              1                                                24  ...   
1              0                                                 1  ...   
2              0                                                 0  ...   
3              1                                                11  ...   
4              1                                                33  ...   

   region-new-england  region-pacific  region-south-atlantic  \
0                   0               0                      0   
1                   0               0                      0   
2                   0               1                      0   
3                   1               0                      0   
4                   0               0                      0   

   region-west-north-central  region-west-south-central  \
0                          0                          0   
1                          0                          0   
2                          0                          0   
3                          0                          0   
4                          0                          0   

   DT-Android Phone / Tablet  DT-MacOS Desktop / Laptop  DT-Other  \
0                          1                          0         0   
1                          0                          1         0   
2                          0                          1         0   
3                          0                          0         0   
4                          0                          0         0   

   DT-Windows Desktop / Laptop  DT-iOS Phone / Tablet  
0                            0                      0  
1                            0                      0  
2                            0                      0  
3                            1                      0  
4                            0                      1  

[5 rows x 50 columns]

###### 4. Gender

In [57]:
df['Gender'].value_counts()

Female    176
Male      158
Name: Gender, dtype: int64

In [58]:
dummy_Gender = pd.get_dummies(df['Gender'])
dummy_Gender.head()

Female  Male
0       0     1
1       0     1
2       0     1
3       0     1
4       0     1

In [59]:
dummy_Gender.rename(columns={'Female':'Gender-Female', 'Male':'Gender-Male'}, inplace=True)
dummy_Gender.head()

Gender-Female  Gender-Male
0              0            1
1              0            1
2              0            1
3              0            1
4              0            1

In [60]:
# concatinating 
df = pd.concat([df,dummy_Gender], axis=1)
# dropping original column aspiration
df.drop('Gender', axis =1, inplace= True)

In [61]:
df.head()

part-time-employed  mental-illness  computer  hospitalized-MI  \
0                   0               0         0                0   
1                   1               1         1                0   
2                   1               0         1                0   
3                   0               0         1                0   
4                   1               1         1                1   

   hospitalized-days  legally-disabled  internet-access  live-with-parents  \
0                0.0                 0                1                  0   
1                0.0                 0                1                  0   
2                0.0                 0                1                  0   
3                3.0                 0                1                  1   
4               35.0                 1                1                  0   

   gap-in-resume  Total length of any gaps in my resume in months.  ...  \
0              1                                                24  ...   
1              0                                                 1  ...   
2              0                                                 0  ...   
3              1                                                11  ...   
4              1                                                33  ...   

   region-south-atlantic  region-west-north-central  \
0                      0                          0   
1                      0                          0   
2                      0                          0   
3                      0                          0   
4                      0                          0   

   region-west-south-central  DT-Android Phone / Tablet  \
0                          0                          1   
1                          0                          0   
2                          0                          0   
3                          0                          0   
4                          0                          0   

   DT-MacOS Desktop / Laptop  DT-Other  DT-Windows Desktop / Laptop  \
0                          0         0                            0   
1                          1         0                            0   
2                          1         0                            0   
3                          0         0                            1   
4                          0         0                            0   

   DT-iOS Phone / Tablet  Gender-Female  Gender-Male  
0                      0              0            1  
1                      0              0            1  
2                      0              0            1  
3                      0              0            1  
4                      1              0            1  

[5 rows x 51 columns]

Great all categorical data is handled.

###### Normalizing Data

In [62]:
df.columns

Index(['part-time-employed', 'mental-illness', 'computer', 'hospitalized-MI',
       'hospitalized-days', 'legally-disabled', 'internet-access',
       'live-with-parents', 'gap-in-resume',
       'Total length of any gaps in my resume in months.',
       'total-annual-income (USD)', 'unemployed', 'read-outside',
       'annual-income-SocialWelfare', 'recieve-food-stamps',
       'housing-section-8', 'hospitalized-for-MI(num.of times)',
       'Lack of concentration', 'Anxiety', 'Depression', 'Obsessive thinking',
       'Mood swings', 'Panic attacks', 'Compulsive behavior', 'Tiredness',
       'Age', 'Household Income', 'edu-completed-masters', 'edu-completed-phd',
       'edu-completed-undergraduate', 'edu-high-school/ged', 'edu-some-phd',
       'edu-some-undergraduate', 'edu-some-highschool', 'Some Masters',
       'region-east-north-central', 'region-east-south-central',
       'region-middle-atlantic', 'region-mountain', 'region-new-england',
       'region-pacific', 'region-sout

###### 1. total-annual-income (USD)

In [63]:
df['total-annual-income (USD)'].value_counts()

0      41
100    25
25     12
50     10
35      9
       ..
78      1
90      1
5       1
48      1
51      1
Name: total-annual-income (USD), Length: 84, dtype: int64

In [64]:
# replace (original value) by (original value)/(maximum value)
df['total-annual-income (USD)'] = df['total-annual-income (USD)']/df['total-annual-income (USD)'].max()

In [65]:
df['total-annual-income (USD)'].value_counts()

0.00    41
1.00    25
0.25    12
0.50    10
0.35     9
        ..
0.78     1
0.90     1
0.05     1
0.48     1
0.51     1
Name: total-annual-income (USD), Length: 84, dtype: int64

###### 2. annual-income-SocialWelfare

In [66]:
# replace (original value) by (original value)/(maximum value)
df['annual-income-SocialWelfare'] = df['annual-income-SocialWelfare']/df['annual-income-SocialWelfare'].max()

In [67]:
df['annual-income-SocialWelfare'].value_counts()

0.00    280
0.01      9
0.02      5
0.12      4
0.10      4
0.04      4
0.14      2
0.25      2
0.30      2
0.20      2
0.50      1
0.83      1
0.24      1
0.05      1
0.44      1
0.23      1
0.45      1
0.13      1
0.59      1
0.17      1
0.54      1
0.11      1
0.60      1
1.00      1
0.07      1
0.93      1
0.03      1
0.66      1
0.47      1
0.06      1
Name: annual-income-SocialWelfare, dtype: int64

###### 3. hospitalized-for-MI(num.of times)

In [68]:
# replace (original value) by (original value)/(maximum value)
df['hospitalized-for-MI(num.of times)'] = df['hospitalized-for-MI(num.of times)']/df['hospitalized-for-MI(num.of times)'].max()

In [69]:
df['hospitalized-for-MI(num.of times)'].value_counts()

0.00    291
0.01     23
0.03      4
0.04      3
0.05      3
0.02      2
0.69      2
0.43      1
0.19      1
0.07      1
0.20      1
1.00      1
0.06      1
Name: hospitalized-for-MI(num.of times), dtype: int64

###### 4. hospitalized-days

In [70]:
# replace (original value) by (original value)/(maximum value)
df['hospitalized-days'] = df['hospitalized-days']/df['hospitalized-days'].max()

In [71]:
df['hospitalized-days'].value_counts()

0.00    255
0.03     38
0.01     13
0.02      3
0.08      2
1.00      2
0.05      2
0.14      2
0.13      1
0.06      1
0.27      1
0.10      1
0.15      1
0.28      1
0.42      1
0.35      1
0.65      1
0.60      1
0.99      1
0.21      1
0.20      1
0.78      1
0.91      1
0.44      1
0.43      1
Name: hospitalized-days, dtype: int64

In [72]:
df.head()

part-time-employed  mental-illness  computer  hospitalized-MI  \
0                   0               0         0                0   
1                   1               1         1                0   
2                   1               0         1                0   
3                   0               0         1                0   
4                   1               1         1                1   

   hospitalized-days  legally-disabled  internet-access  live-with-parents  \
0               0.00                 0                1                  0   
1               0.00                 0                1                  0   
2               0.00                 0                1                  0   
3               0.03                 0                1                  1   
4               0.35                 1                1                  0   

   gap-in-resume  Total length of any gaps in my resume in months.  ...  \
0              1                                                24  ...   
1              0                                                 1  ...   
2              0                                                 0  ...   
3              1                                                11  ...   
4              1                                                33  ...   

   region-south-atlantic  region-west-north-central  \
0                      0                          0   
1                      0                          0   
2                      0                          0   
3                      0                          0   
4                      0                          0   

   region-west-south-central  DT-Android Phone / Tablet  \
0                          0                          1   
1                          0                          0   
2                          0                          0   
3                          0                          0   
4                          0                          0   

   DT-MacOS Desktop / Laptop  DT-Other  DT-Windows Desktop / Laptop  \
0                          0         0                            0   
1                          1         0                            0   
2                          1         0                            0   
3                          0         0                            1   
4                          0         0                            0   

   DT-iOS Phone / Tablet  Gender-Female  Gender-Male  
0                      0              0            1  
1                      0              0            1  
2                      0              0            1  
3                      0              0            1  
4                      1              0            1  

[5 rows x 51 columns]

###### Assigning categories to Age and Household income for better analysis for strong predictor.

**Assigning low, medium and high categories to the salery ranges.**

In [73]:
df['Household Income'].value_counts()

$25,000-$49,999         68
$50,000-$74,999         57
Prefer not to answer    38
$10,000-$24,999         35
$75,000-$99,999         33
$0-$9,999               27
$100,000-$124,999       24
$200,000+               20
$125,000-$149,999       16
$150,000-$174,999       14
$175,000-$199,999        2
Name: Household Income, dtype: int64

In [74]:
df['Household Income'].replace('Prefer not to answer','$175,000-$199,999', inplace=True)

In [75]:
df['Household Income'].value_counts()

$25,000-$49,999      68
$50,000-$74,999      57
$175,000-$199,999    40
$10,000-$24,999      35
$75,000-$99,999      33
$0-$9,999            27
$100,000-$124,999    24
$200,000+            20
$125,000-$149,999    16
$150,000-$174,999    14
Name: Household Income, dtype: int64

In [76]:
# Mapping dictionary for income categories
income_mapping = {
    '$0-$9,999': 'Low',
    '$10,000-$24,999': 'Low',
    '$25,000-$49,999': 'Medium',
    '$50,000-$74,999': 'Medium',
    '$75,000-$99,999': 'Medium',
    '$100,000-$124,999': 'High',
    '$125,000-$149,999': 'High',
    '$150,000-$174,999': 'High',
    '$175,000-$199,999': 'High',
    '$200,000+': 'High'
}

In [77]:
# Function to assign income categories
def assign_income_category(income_range):
    return income_mapping.get(income_range) 

In [78]:
df['HHI_Category'] = df['Household Income'].apply(assign_income_category)

In [79]:
df['Household Income'].head()

0      $25,000-$49,999
1      $50,000-$74,999
2    $150,000-$174,999
3      $25,000-$49,999
4      $25,000-$49,999
Name: Household Income, dtype: object

In [80]:
df.head()

part-time-employed  mental-illness  computer  hospitalized-MI  \
0                   0               0         0                0   
1                   1               1         1                0   
2                   1               0         1                0   
3                   0               0         1                0   
4                   1               1         1                1   

   hospitalized-days  legally-disabled  internet-access  live-with-parents  \
0               0.00                 0                1                  0   
1               0.00                 0                1                  0   
2               0.00                 0                1                  0   
3               0.03                 0                1                  1   
4               0.35                 1                1                  0   

   gap-in-resume  Total length of any gaps in my resume in months.  ...  \
0              1                                                24  ...   
1              0                                                 1  ...   
2              0                                                 0  ...   
3              1                                                11  ...   
4              1                                                33  ...   

   region-west-north-central  region-west-south-central  \
0                          0                          0   
1                          0                          0   
2                          0                          0   
3                          0                          0   
4                          0                          0   

   DT-Android Phone / Tablet  DT-MacOS Desktop / Laptop  DT-Other  \
0                          1                          0         0   
1                          0                          1         0   
2                          0                          1         0   
3                          0                          0         0   
4                          0                          0         0   

   DT-Windows Desktop / Laptop  DT-iOS Phone / Tablet  Gender-Female  \
0                            0                      0              0   
1                            0                      0              0   
2                            0                      0              0   
3                            1                      0              0   
4                            0                      1              0   

   Gender-Male  HHI_Category  
0            1        Medium  
1            1        Medium  
2            1          High  
3            1        Medium  
4            1        Medium  

[5 rows x 52 columns]

In [81]:
df['HHI_Category'].value_counts()

Medium    158
High      114
Low        62
Name: HHI_Category, dtype: int64

In [82]:
dummy_hhi_cat = pd.get_dummies(df['HHI_Category'])
dummy_hhi_cat.head()

High  Low  Medium
0     0    0       1
1     0    0       1
2     1    0       0
3     0    0       1
4     0    0       1

In [83]:
dummy_hhi_cat.rename(columns={'High':'hhi-high', 'Low':'hhi-low','Medium':'hhi-medium'}, inplace=True)
dummy_hhi_cat.head()

hhi-high  hhi-low  hhi-medium
0         0        0           1
1         0        0           1
2         1        0           0
3         0        0           1
4         0        0           1

In [84]:
# concatinating 
df = pd.concat([df,dummy_hhi_cat], axis=1)
# dropping original column aspiration
df.drop('HHI_Category', axis =1, inplace= True)

In [85]:
df.head()

part-time-employed  mental-illness  computer  hospitalized-MI  \
0                   0               0         0                0   
1                   1               1         1                0   
2                   1               0         1                0   
3                   0               0         1                0   
4                   1               1         1                1   

   hospitalized-days  legally-disabled  internet-access  live-with-parents  \
0               0.00                 0                1                  0   
1               0.00                 0                1                  0   
2               0.00                 0                1                  0   
3               0.03                 0                1                  1   
4               0.35                 1                1                  0   

   gap-in-resume  Total length of any gaps in my resume in months.  ...  \
0              1                                                24  ...   
1              0                                                 1  ...   
2              0                                                 0  ...   
3              1                                                11  ...   
4              1                                                33  ...   

   DT-Android Phone / Tablet  DT-MacOS Desktop / Laptop  DT-Other  \
0                          1                          0         0   
1                          0                          1         0   
2                          0                          1         0   
3                          0                          0         0   
4                          0                          0         0   

   DT-Windows Desktop / Laptop  DT-iOS Phone / Tablet  Gender-Female  \
0                            0                      0              0   
1                            0                      0              0   
2                            0                      0              0   
3                            1                      0              0   
4                            0                      1              0   

   Gender-Male  hhi-high  hhi-low  hhi-medium  
0            1         0        0           1  
1            1         0        0           1  
2            1         1        0           0  
3            1         0        0           1  
4            1         0        0           1  

[5 rows x 54 columns]

**Writing a function to assign categories names to the ranges of age.**
1. 18-29/30-44 years - I have called this group young adults
2. 45-60 years - I have called this group adults
3. Greater than 60 years - I have called this group seniors

In [86]:
df['Age'].value_counts()

30-44    103
45-60     99
> 60      80
18-29     52
Name: Age, dtype: int64

In [87]:
# Function to assing categories 
def assign_age_category(age_range):
    if age_range in ['30-44', '18-29']:
        return 'Young Adults'
    elif age_range == '45-60':
        return 'Adults'
    elif age_range == '> 60':
        return 'Seniors'
    else:
        return 'Other'  # Handle other cases if needed

In [88]:
# Apply the function to create the 'Age_Category' column
df['Age_Category'] = df['Age'].apply(assign_age_category)

In [89]:
df['Age_Category'].head()

0    Young Adults
1    Young Adults
2    Young Adults
3    Young Adults
4    Young Adults
Name: Age_Category, dtype: object

In [90]:
df.head()

part-time-employed  mental-illness  computer  hospitalized-MI  \
0                   0               0         0                0   
1                   1               1         1                0   
2                   1               0         1                0   
3                   0               0         1                0   
4                   1               1         1                1   

   hospitalized-days  legally-disabled  internet-access  live-with-parents  \
0               0.00                 0                1                  0   
1               0.00                 0                1                  0   
2               0.00                 0                1                  0   
3               0.03                 0                1                  1   
4               0.35                 1                1                  0   

   gap-in-resume  Total length of any gaps in my resume in months.  ...  \
0              1                                                24  ...   
1              0                                                 1  ...   
2              0                                                 0  ...   
3              1                                                11  ...   
4              1                                                33  ...   

   DT-MacOS Desktop / Laptop  DT-Other  DT-Windows Desktop / Laptop  \
0                          0         0                            0   
1                          1         0                            0   
2                          1         0                            0   
3                          0         0                            1   
4                          0         0                            0   

   DT-iOS Phone / Tablet  Gender-Female  Gender-Male  hhi-high  hhi-low  \
0                      0              0            1         0        0   
1                      0              0            1         0        0   
2                      0              0            1         1        0   
3                      0              0            1         0        0   
4                      1              0            1         0        0   

   hhi-medium  Age_Category  
0           1  Young Adults  
1           1  Young Adults  
2           0  Young Adults  
3           1  Young Adults  
4           1  Young Adults  

[5 rows x 55 columns]

In [91]:
df['Age_Category'].value_counts()

Young Adults    155
Adults           99
Seniors          80
Name: Age_Category, dtype: int64

In [92]:
dummy_age_cat = pd.get_dummies(df['Age_Category'])
dummy_age_cat.head()

Adults  Seniors  Young Adults
0       0        0             1
1       0        0             1
2       0        0             1
3       0        0             1
4       0        0             1

In [93]:
dummy_age_cat.rename(columns={'Adults':'age-adults', 'Seniors':'age-seniors','Young Adults':'age-young-adults'}, inplace=True)
dummy_age_cat.head()

age-adults  age-seniors  age-young-adults
0           0            0                 1
1           0            0                 1
2           0            0                 1
3           0            0                 1
4           0            0                 1

In [94]:
# concatinating 
df = pd.concat([df,dummy_age_cat], axis=1)
# dropping original column aspiration
df.drop('Age_Category', axis =1, inplace= True)

In [95]:
df.head()

part-time-employed  mental-illness  computer  hospitalized-MI  \
0                   0               0         0                0   
1                   1               1         1                0   
2                   1               0         1                0   
3                   0               0         1                0   
4                   1               1         1                1   

   hospitalized-days  legally-disabled  internet-access  live-with-parents  \
0               0.00                 0                1                  0   
1               0.00                 0                1                  0   
2               0.00                 0                1                  0   
3               0.03                 0                1                  1   
4               0.35                 1                1                  0   

   gap-in-resume  Total length of any gaps in my resume in months.  ...  \
0              1                                                24  ...   
1              0                                                 1  ...   
2              0                                                 0  ...   
3              1                                                11  ...   
4              1                                                33  ...   

   DT-Windows Desktop / Laptop  DT-iOS Phone / Tablet  Gender-Female  \
0                            0                      0              0   
1                            0                      0              0   
2                            0                      0              0   
3                            1                      0              0   
4                            0                      1              0   

   Gender-Male  hhi-high  hhi-low  hhi-medium  age-adults  age-seniors  \
0            1         0        0           1           0            0   
1            1         0        0           1           0            0   
2            1         1        0           0           0            0   
3            1         0        0           1           0            0   
4            1         0        0           1           0            0   

   age-young-adults  
0                 1  
1                 1  
2                 1  
3                 1  
4                 1  

[5 rows x 57 columns]

###### Formatting Data types

In [96]:
df.dtypes

part-time-employed                                    int64
mental-illness                                        int64
computer                                              int64
hospitalized-MI                                       int64
hospitalized-days                                   float64
legally-disabled                                      int64
internet-access                                       int64
live-with-parents                                     int64
gap-in-resume                                         int64
Total length of any gaps in my resume in months.      int64
total-annual-income (USD)                           float64
unemployed                                            int64
read-outside                                          int64
annual-income-SocialWelfare                         float64
recieve-food-stamps                                   int64
housing-section-8                                     int64
hospitalized-for-MI(num.of times)       

In [97]:
# age and house hold income have data type object which is not good for computation as i transformed these columns,
# so going to remove these columns
df.drop('Age', axis =1, inplace= True)
df.drop('Household Income', axis =1, inplace= True)

In [98]:
df.columns

Index(['part-time-employed', 'mental-illness', 'computer', 'hospitalized-MI',
       'hospitalized-days', 'legally-disabled', 'internet-access',
       'live-with-parents', 'gap-in-resume',
       'Total length of any gaps in my resume in months.',
       'total-annual-income (USD)', 'unemployed', 'read-outside',
       'annual-income-SocialWelfare', 'recieve-food-stamps',
       'housing-section-8', 'hospitalized-for-MI(num.of times)',
       'Lack of concentration', 'Anxiety', 'Depression', 'Obsessive thinking',
       'Mood swings', 'Panic attacks', 'Compulsive behavior', 'Tiredness',
       'edu-completed-masters', 'edu-completed-phd',
       'edu-completed-undergraduate', 'edu-high-school/ged', 'edu-some-phd',
       'edu-some-undergraduate', 'edu-some-highschool', 'Some Masters',
       'region-east-north-central', 'region-east-south-central',
       'region-middle-atlantic', 'region-mountain', 'region-new-england',
       'region-pacific', 'region-south-atlantic', 'region-west-n

Great columns are removed and now data is ready for Feature Engineering to figure out strong predictors for mental health prediction.

In [99]:
# saving current file as cleaned file
#df.to_csv('cleaned_df.csv')